### DISCUSSION TITLE
Reflect postgresql `dialect_options` for indexes

### DISCUSSION THREAD

**Discussion by chrisjsewell** [sqlalchemy/sqlalchemy#7698] [Usage Questions] [answered]
Heya, I have an index on a table that looks like:

```python
Index(
  'my_index',
  'column_name',
  postgresql_using='btree',
  postgresql_ops={'column_name': 'varchar_pattern_ops'},
)
```

which generates `CREATE INDEX my_index ON public.my_table USING btree (column_name varchar_pattern_ops)`

When I use `inspect(engine).get_indexes('my_table')`, I get `[{'name': 'my_index', 'unique': False, 'column_names': ['column_name'], 'include_columns': []}]`, but would like to get info e.g. that `varchar_pattern_ops` was used.

Am I missing something, or is this just not supported at present?

**Comment by zzzeek** [comment, marked as answer]
looks like PG index reflection currently supports "postgresql_include", "postgresql_with", "postgresql_using", "postgresql_where", and "column_sorting".   no "postgresql_ops" right now.

**Reply by chrisjsewell** [reply]
thanks, when you say it supports `postgresql_using`, how do you retrieve that information?

**Reply by zzzeek** [reply]
I read the source code here:

https://github.com/sqlalchemy/sqlalchemy/blob/d6b3c82b0c329730bcaff42b4bb39dba83acb536/lib/sqlalchemy/dialects/postgresql/base.py#L4329-L4362

please note I am not implying that **you** should be reading the source code to find this out.    As far as whether we would "document" this, depending on how this information can be read, I'd rather support the feature instead of documenting that it isn't supported.  Usually when we document something isn't supported, later on someone contributes it and nobody knows to change that documentation.   

at the same time there are some other long term limitations on PG index reflection also, like reflection of function-based indexes, that also doesnt seem to have any docs.   it's likely best to note these things.

**Reply by chrisjsewell** [reply]
Oh thats fine, I'm happy to dig through the source code, when I'm not being lazy lol.

FYI, this is why `postgresql_using` is not showing:

https://github.com/sqlalchemy/sqlalchemy/blob/d6b3c82b0c329730bcaff42b4bb39dba83acb536/lib/sqlalchemy/dialects/postgresql/base.py#L4318-L4323
